edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/Avaliability.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Availability
+    {
+        private readonly ISystemTime time;
+
+        public string Name;
+        public string Version;
+        public TimeSpan TotalTime = TimeSpan.Zero;
+        public TimeSpan Uptime = TimeSpan.Zero;
+
+        private DateTime? previousMeasure = null;
+
+        public Availability(string name, string version, ISystemTime time)
+        {
+            this.Name = Preconditions.CheckNotNull(name, nameof(name));
+            this.Version = Preconditions.CheckNotNull(version, nameof(version));
+            this.time = Preconditions.CheckNotNull(time, nameof(time));
+            this.previousMeasure = time.UtcNow;
+        }
+
+        public Availability(AvailabilityRaw raw, ISystemTime time)
+        {
+            this.Name = Preconditions.CheckNotNull(raw.Name);
+            this.Version = Preconditions.CheckNotNull(raw.Version);
+            this.Uptime = Preconditions.CheckNotNull(raw.Uptime);
+            this.TotalTime = Preconditions.CheckNotNull(raw.TotalTime);
+
+            this.time = Preconditions.CheckNotNull(time, nameof(time));
+        }
+
+        public double AvailabilityRatio
+        {
+            get
+            {
+                if (this.TotalTime == TimeSpan.Zero)
+                {
+                    return 0;
+                }
+
+                return this.Uptime.TotalMilliseconds / this.TotalTime.TotalMilliseconds;
```

Comment:
Please have a C# handle case of Divide by zero here. ( Ideally this should never happend but who knows )

Reply:
see if statement above

Reply:
Ohhhh I see. thanks!